tests/distributions/test_truncated.py

```diff
@@ -400,6 +404,28 @@ def test_truncated_inference():
     assert np.isclose(map["lam"], lam_true, atol=0.1)
 
 
+def test_truncated_normal_logcdf_inference():
+    lower = 0
+    upper = 1
+
+    rng = np.random.default_rng(260)
+    x = rng.normal(0, size=5000)
+    obs = x[np.where(~((x < lower) | (x > upper)))]
+
+    with Model() as m:
+        mu = Normal("mu", 10, 1)
+        sigma = Uniform("sigma", 1, 10)
+        normal_ = Normal.dist(mu, sigma)
+        trunc_normal = Truncated.dist(normal_, lower=0)
+        censored = Censored("censored", trunc_normal, lower=None, upper=upper, observed=obs)
+
+    ip = m.initial_point()
+    norm_logp, trunc_logp, cens_logp = m.compile_logp(sum=False)(ip)
+    assert np.all(np.isfinite(norm_logp))
+    assert np.all(np.isfinite(trunc_logp))
+    assert np.all(np.isfinite(cens_logp))
```

Comment:
This test is a bit too indirect. We should just check the logcdf in the `TruncatedNormal` like we test other distributions logcdf in `tests/distributions/continuous.py`

Reply:
Hey so I've tried this test instead, 
```python
from pymc.testing import (
    R,
    Rplus,
    Rminusbig,
    check_logcdf,
    select_by_precision,
)
import pymc as pm
def test_trunc_norm_logcdf():
        check_logcdf(
            pm.TruncatedNormal,
            R,
            {"mu": R, "sigma": Rplus, 'lower': Rminusbig, 'upper':Rplus},
            lambda value, mu, sigma, lower, upper: scipy.stats.truncnorm((lower - mu) / sigma, (upper - mu) / sigma, mu, sigma,).logcdf(value),
            decimal=select_by_precision(float64=5, float32=1),
        )
```
But I'm getting an assertion error:
`
AssertionError: 
Arrays are not almost equal to 5 decimals
{'mu': array(-2.1), 'sigma': array(0.01), 'lower': array(-1.), 'upper': array(1.5), 'value': array(0.01)}
x and y nan location mismatch:
 x: array(nan)
 y: array(-0.)
`
I'm not sure if this is just a factor of the test conditions I've set or whether there is actually something wrong/different with the logcdf implementation vs. scipy....

Reply:
That's too extreme, mu is -2.1, and sigma is only 0.01, and we are testing logp at value 0.01

Reply:
PyMC is returning `nan` and scipy `-0` I believe. I had a look in `test_continuous` and realised there were a similar set of tests for `TruncatedNormal` logp which set `skip_paramdomain_outside_edge_test = True` and use different domains, so I have used the same params and the logcdf tests pass. I've pushed this so let me kow if you're happy with the new tests.

Reply:
Can you check where the nan is coming from by breaking apart the pytensor expression? I imagine there is a -inf - -inf or something, but not clear where from? Asking in case there is a specific edge case we can patch to avoid nan

Reply:
I found it and fixed the bug, the test with the domains mentioned in the comment above now pass.